cal/cal/static/js/app.js

```diff
@@ -3,109 +3,171 @@ var analyticsApp = angular.module('analyticsApp', ['nvd3', 'ui.calendar']);
-            editing: false
+  $scope.$on('calendarRange:updated', function(event, data) {
+    var rangeData = CalendarRangeService.getRange();
+    var timeRange;
+    if (_this.isCumulative) {
+      _this.updateTags('cumulative', null, null);
+    } else {
+      _this.updateTags(rangeData.timeRange, rangeData.start, rangeData.end)
+    }
+  });
+
+  this.updateTags = function updateTags(timeRange, start, end) {
+    _this.timeRange = timeRange;
+    $http({
+      method: 'GET',
+      url: tagUrl + '.json',
+      cache: true,
+      params: {
+        start: (start)? start.toISOString() : null,
+        end: (end)? end.toISOString() : null,
+      }
+    })
+      .success(function successCallback(data) {
+        _this.tags[timeRange] = [];
+        for (var i = 0; i < data.results.length; i++) {
+          var tag = data.results[i];
+          _this.tags[timeRange].push({
+            id: tag.id,
+            label: tag.label,
+            keywords: tag.keywords,
+            hours: tag.hours
           });
-        });
-    };
-
-    this.startEdit = function(tagId) {
-      var tag = $scope.tags.find(function(tag, index, array) { return tag.id == tagId; });
-      tag.newLabel = tag.label;
-      tag.newKeywords = tag.keywords;
-      tag.editing = true;
-    };
-
-    this.submit = function(tagId) {
-      var tag = $scope.tags.find(function(tag, index, array) { return tag.id == tagId; });
-      tag.editing = false;
-
-      $http({
-        method: 'POST',
-        url: tagUrl + '/' + tagId,
-        data: $.param({
-          label: tag.newLabel,
-          keywords: tag.newKeywords,
-          csrfmiddlewaretoken: getCookie('csrftoken'),
-          _method: 'PATCH'
-        }),
-        headers: {
-          'Content-Type': 'application/x-www-form-urlencoded'
-        }
-      }).
-        success(function addToList(data) {
-          tag.label = data.label;
-          tag.keywords = data.keywords;
-          tag.hours = data.hours;
-        });
-    };
-
-    this.cancelEdit = function(tagId) {
-      var tag = $scope.tags.find(function(tag, index, array) { return tag.id == tagId; });
-      tag.editing = false;
-    };
-
-    this.delete = function(tagId) {
-      $http({
-        method: 'POST',
-        url: tagUrl + '/' + tagId,
-        data: $.param({
-          csrfmiddlewaretoken: getCookie('csrftoken'),
-          _method: 'DELETE'
-        }),
-        headers: {
-          'Content-Type': 'application/x-www-form-urlencoded'
         }
-      }).
-        success(function removeFromList(data) {
-          $scope.tags = $scope.tags.filter(function(tag) {
-            return tag.id !== tagId;
-          });
-        });
-    };
+      });
+  }
+  var initialTimeRange = CalendarRangeService.getRange()
+  this.updateTags(initialTimeRange.timeRange, initialTimeRange.start, initialTimeRange.end);
+
+  this.hideZeroHoursFilter = function (value, index, array) {
+    if (this.hideZeroHours && value.hours === 0) {
```

Comment:
I think the expanded function is easier to read and understand, and is more in line with the semantics of the logic. Thoughts?